### ISSUE TITLE
Display wide widgets as modals in Customizer

### ISSUE THREAD

**Issue by critterverse** [WordPress/gutenberg#32210]
Hi all, this is an alternate solution to the issue being explored in #31736. We originally tried to use the popover component to display wide widgets in the Customizer, but it's proving to be difficult to wrangle in terms of positioning, size, etc.

A modal solution might provide a more consistent user experience:

![Customizer_XLwidgets](https://user-images.githubusercontent.com/77359364/119529271-f1b26b00-bd4f-11eb-985d-a5877f6535ec.jpg)

Using the move arrows could be similar to reordering nav items from the “block navigation” action in the Navigation block toolbar, where the items move around behind the overlay:

https://user-images.githubusercontent.com/77359364/119529301-f9720f80-bd4f-11eb-8f64-998909a27ef2.mov

I think any actions from the ellipses menu would close the modal and take place in the sidebar while the legacy widget is in preview mode.

Thoughts on this direction?

**Comment by tellthemachines** [comment]
Thanks for the exploration @critterverse ! This is something we may be able to look at in the future, but is too complex to try implementing for this release, given that we're past feature freeze and getting close to Beta 1. 

The main technical problem here is how to move the block toolbar into a modal. The block toolbar component sits outside the content area of the editor, and we can't control where it renders from inside the block, so we'd need to modify the actual toolbar component to support that behaviour.

**Comment by stokesman** [comment]
This solution is creative. Technical challenges aside, I don't see using the modal as advancing the interface as compared to what the Customizer (before/without the block editor) provides for wide widgets. I do believe it's generally agreed that what we have now should be improved upon. So to that end I've explored a couple of ways to bring the wide widget display closer to the pre-block-editor interface, #33230 and #33236. The former is an approach to improve the popover positioning. The latter is also that (implemented differently) plus a few more changes as proposed improvements.

I don't know how much a priority any of this would be so I'm not urging anyone to try them out but merely marking their existence.

**Comment by noisysocks** [comment]
I agree with @stokesman that improving what we already have is likely more practical (from a cost benefit point of view) than a new modal interface.
